### PR TITLE
Remove unnecessary test dependencies

### LINKED ISSUE #15686: Incremental build improvements
**Description**
Incremental PR builds seem to be running smoothly (and much faster, on average) but there is even more potential for improvements. This issue shall track those ideas.

Most of those ideas came up here: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Incremental.20build

**Implementation ideas**
- [ ] CI: execute `integration-tests/gradle` via Maven with enabled GIB
  Needs script-supported mirroring of the dependencies in all gradle build files in that module.
  
- [ ] Reduce the number of extensions used by `integration-tests/gradle` (see also [comment](https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Incremental.20build/near/228432916) by @glefloch)
  
- [ ] Mostly CI, but could also help locally: Wrap clone and execution of `resteasy-reactive-testsuite` in e.g. `tcks/resteasy-reactive`, with `exec-maven-plugin` and those few deps the testsuite is (indirectly) dependening on
  
- [ ] CI: Don't execute native jobs at all in case the respective modules are not at all "impacted" (GIB-speak).
  This could work by executing `mvn validate -Dquickly-ci ... -Dgib.logImpactedTo=...` in some (new) upstream job (probably after the initial build job) which is then evaluating the logfile to prune the native jobs json file.
  
- [ ] CI: in case the previous works, leverage that geenral approach to other jobs that execute only a fraction of modules (or even only one, e.g. `integration-tests/maven`).
  Instead of introducing matrices for all jobs, a simple `if` at job level might be more appropriate.
  Note: In theory, jobs without any `-pl ...` or with `-pl !...` could benefit from this as well, but the projects list to pass to Maven could be _huge_ (> 800 for a full build).
  
- [ ] CI: build `main` branch incrementally (has to be discussed carefully first!)
  Requires a "moving" tag as a clear reference and maybe a "mirroring" full job, nightly or weekly.
  
- [ ] ...?

### PR DESCRIPTION
**Pull request by glefloch**
This branch leverage all dependencies used in gradle test IT. Addresses task number 2 from #15686.

@famod, let's see how build goes :). I manually updated the `pom.xml`. 
In the `pom.xml`, there are some `-deployment` dependencies that are not used in any gradle project (like `quarkus-grpc-deployment`, `quarkus-kotlin-deployment`) should we keep them?

### CODE REVIEW

**Comment by famod** [comment]
> I manually updated the `pom.xml`.

Don't trust my script? 😄 

> In the `pom.xml`, there are some `-deployment` dependencies that are not used in any gradle project (like `quarkus-grpc-deployment`, `quarkus-kotlin-deployment`) should we keep them?

If the runtime counterparts are used: yes absolutely! For the same reason we enrich all other ITs with those minimal deployment dependencies: To ensure Maven builds the deployment module(s) before the IT module.

**Comment by famod** [comment]
GH actions are broken ATM: https://www.githubstatus.com/incidents/s654n76c1bwr

File: integration-tests/gradle/src/test/resources/basic-java-platform-module/build.gradle
Comment (famod) [inline integration-tests/gradle/src/test/resources/basic-java-platform-module/build.gradle:21]: Ah this reminds me of something I wanted to ask you anyway: In CI we use `.github/mvn-settings.xml` for Maven, which uses the Google EU central mirror, but I have seen Gradle tests downloading stuff from central instead.
So I guess something is missing to pass it to Gradle in CI?

**Comment by famod** [comment]
I've just rebased this branch to trigger CI now that GH Actions are back to normal. I hope you don't mind. 🙂

File: integration-tests/gradle/src/test/resources/basic-java-platform-module/build.gradle
Comment (glefloch) [reply integration-tests/gradle/src/test/resources/basic-java-platform-module/build.gradle:21]: Actually, I don't know, if this is a central mirror, I think this could be ok. I will add it too, in case of `mavenCentral()` failure, it will be used.

**Review by famod — APPROVED** [review]
LGTM!

Not sure if anyone else should have a look before merging. I'll leave this up to you @glefloch.

**Review by gsmet — APPROVED** [review]
This is very useful, thanks for taking the time to do that!

**Pull request merged**
